Makefile

```diff
@@ -140,9 +141,21 @@ deploy: manifests kustomize ## Deploy controller to the K8s cluster specified in
 undeploy: ## Undeploy controller from the K8s cluster specified in ~/.kube/config.
 	$(KUSTOMIZE) build config/default | kubectl delete -f -
 
+$(RELEASE_DIR):
+	mkdir -p $(RELEASE_DIR)
+
+publish-cluster-templates: $(RELEASE_DIR) cluster-templates
+	cp $(BYOH_TEMPLATES)/v1beta1/cluster-template.yaml $(RELEASE_DIR)/cluster-template.yaml
+
 publish-infra-yaml:kustomize # Generate infrastructure-components.yaml for the provider
 	cd config/manager && $(KUSTOMIZE) edit set image gcr.io/k8s-staging-cluster-api/cluster-api-byoh-controller=${IMG}
-	$(KUSTOMIZE) build config/default > infrastructure-components.yaml
+	$(KUSTOMIZE) build config/default > $(RELEASE_DIR)/infrastructure-components.yaml
+
+publish-metadata-yaml:
+	cp metadata.yaml $(RELEASE_DIR)/metadata.yaml
+
+publish-host-agent-binary: host-agent-binaries
```

Comment:
This should be `build-host-agent-binary`? 

Because all of these targets are not actually publishing anything.
`publish-cluster-templates` > `build-cluster-templates`
`publish-infra-yaml` > `build-infra-yaml`
`publish-metadata-yaml` > `build-metadata-yaml`

And now that all of these assets will be copied to `RELEASE_DIR`. There will be one more target which says's

`publish-draft-release` and that will actually push content as a draft release (can be accommodated in https://github.com/vmware-tanzu/cluster-api-provider-bringyourownhost/issues/181). 

@anusha94 WDYT?

Reply:
I have added these targets here https://github.com/vmware-tanzu/cluster-api-provider-bringyourownhost/pull/261

Reply:
@shamsher31 Awesome. I'll close this PR then!